Makefile

```diff
@@ -134,7 +134,7 @@ test-unit: use-main
 	@go test -v -race $(shell go list ./... | grep -v tests/)
 
 test-integration: tidy
-	@go test -v -race ./tests/integration/... 
```

Comment:
😢

Reply:
Can we open an issue to track this fix--either on our side or in cosmos-sdk. @nivasan1 maybe has more context on exactly what is causing the race detector to trigger.